src/main/java/frc/robot/subsystems/drive/states/LineupState.java

```diff
@@ -151,99 +160,128 @@ public Rotation2d getRotationForReefSide() {
     }
   }
 
-  /**
-   * checks if lineup is within 0.01 of tag + offsets
-   *
-   * @return true if error is small enoguh
-   */
-  public boolean lineupFinished() {
-    return latestObservation != null
-        && (latestObservation.alongTrackDistance() < 0.05
-            && Math.abs(latestObservation.crossTrackDistance()) < 0.01);
-  }
-
-  public void periodic() {
-    // run test through here if in test mode
-    if (DriverStation.isTest()) {
-      this.testPeriodic();
-    }
-    this.LineupWithReefLocation();
-
-    drive.setDriveLinedUp(lineupFinished());
-    Logger.recordOutput("lineup/finished", lineupFinished());
-  }
-
-  /**
-   * gets tag to use for final alignment with vision
-   *
-   * @return int representing tag id to use
-   */
-  public int getTagIdForReef() {
-    boolean allianceRed = drive.isAllianceRed();
+  public boolean checkForSideSwitch() {
     switch (drive.getDesiredLocation()) {
       case Reef0:
+        if (lastReefLocation == DesiredLocation.Reef1) {
+          return true;
+        }
+        return false;
```

Comment:
Please update to `return lastReefLocation == DesiredLocation.Reef1;` here and below (changing the reef ID of course) @linglejack06